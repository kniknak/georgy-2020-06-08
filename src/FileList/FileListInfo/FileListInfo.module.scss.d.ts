declare const styles: {
    readonly root: string;
    readonly count: string;
    readonly size: string;
};
export = styles;