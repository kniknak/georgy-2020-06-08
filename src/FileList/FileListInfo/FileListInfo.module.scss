@mixin mobile {
    @media only screen and (max-width: 375px) {
        @content;
    }
}

.root {
    display: flex;
    padding: 12px;
    margin-bottom: 12px;
    align-items: center;

    @include mobile {
        flex-wrap: wrap;
        align-items: baseline;
    }
}

.count {
    white-space: nowrap;
    margin-right: 12px;
    font-size: 24px;

    @include mobile {
        width: 100%;
        margin-right: 0;
        margin-bottom: 12px;
    }
}

.size {
    white-space: nowrap;
    margin-left: auto;

    @include mobile {
        width: 100%;
    }
}
