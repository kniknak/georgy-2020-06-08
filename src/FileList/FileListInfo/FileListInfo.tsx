import React from "react";
import * as styles from "./FileListInfo.module.scss";
import { formatSize } from "../formatSize";
import { FileInfo } from "../../api/types";

export interface FileListInfoProps {
    files: FileInfo[]
}

export const FileListInfo: React.FC<FileListInfoProps> = ({ files }) => {
    const fileSize = files.reduce((size, file) => size + file.size, 0);

    return (
        <div className={styles.root}>
            <div className={styles.count}>{files.length} Documents</div>
            <div className={styles.size}>Total size: {formatSize(fileSize)}</div>
        </div>
    );
};

