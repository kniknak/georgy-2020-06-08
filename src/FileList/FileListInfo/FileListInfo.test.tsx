import React from "react";
import { shallow } from "enzyme";
import { FileListInfo, FileListInfoProps } from "./FileListInfo";

describe("FileList", () => {
    describe("FileListInfo", () => {
        const defaultProps: FileListInfoProps = {
            files: [{
                originalName: "foo",
                size: 42,
                _id: "bar",
                createdAt: 200,
            }],
        };

        it("renders", () => {
            const subject = shallow(<FileListInfo {...defaultProps}/>);

            expect(subject).toMatchSnapshot();
        });
    });
});