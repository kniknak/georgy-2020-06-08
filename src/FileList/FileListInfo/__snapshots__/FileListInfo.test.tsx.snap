// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList FileListInfo renders 1`] = `
<div>
  <div>
    1
     Documents
  </div>
  <div>
    Total size: 
    42b
  </div>
</div>
`;
